### PR TITLE
343 links publicly searchable fix

### PR DESCRIPTION
**Pull request by glenn-louarn**
This pull request make links publicly searchable, see #343
It also include description on links #342 as the search feature also search in description.

### CODE REVIEW

**Comment by glenn-louarn** [comment]
This PR resolve all conflict and is ready for merge

File: server/handlers/auth.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ const authenticate = (
     passport.authenticate(type, (err, user) => {
       if (err) return next(err);
 
-      if (!user && isStrict) {
+      if (!env.SEARCH_ENABLED && !user && isStrict) {
```
Comment (poeti8) [inline server/handlers/auth.ts:28]: This is a big auth issue, if SEARCH_ENABLED is true everyone can pass auth and access handlers.

**Comment by poeti8** [comment]
1. Use `npm run migrate:make is_searchable` to create a migration file for `isSearchable`. Example: https://github.com/thedevs-network/kutt/blob/develop/server/migrations/20200718124944_description.ts

2. Since the eye icon is just an indicator and should not be clickable it can have the same style as "password protected" indicator or "banned" icon:
image

3. Is searchable checkbox has too much space:
image

4. The style looks very out of place and also nothing happens when I press enter since the submit is not being handled properly. If showing to non-logged-in users is not a concern I'd suggest to add a checkbox to the main table and allow users to search there.
image

image

File: server/models/link.ts [hunk 2]
```diff
@@ -10,6 +10,10 @@ export async function createLinkTable(knex: Knex) {
       table.increments("id").primary();
       table.string("address").notNullable();
       table.string("description");
+      table
+        .boolean("isSearchable")
```
Comment (poeti8) [inline server/models/link.ts:14]: to be consistent with other fields please use `searchable`.

**Comment by glenn-louarn** [comment]
Here are all the modifications you requested, for the search page I improved its design.

**Comment by glenn-louarn** [comment]
Hello, 
I have resolve all conflicts, could you please review the PR ?

**Pull request closed without merging**